src/main/java/seedu/organizer/model/Organizer.java

```diff
@@ -156,17 +168,53 @@ public void addTag(Tag t) throws UniqueTagList.DuplicateTagException {
         tags.add(t);
     }
 
+    /**
+     * Removes {@code tag} from {@code task} in this {@code Organizer}.
+     * @throws TaskNotFoundException if the {@code task} is not in this {@code Organizer}.
+     */
+    private void removeTagFromTask(Tag tag, Task task) throws TaskNotFoundException {
+        Set<Tag> newTags = new HashSet<>(task.getTags());
+
+        if (!newTags.remove(tag)) {
+            return;
+        }
+
+        Task newTask =
+                new Task(task.getName(), task.getPhone(), task.getEmail(), task.getAddress(), newTags);
+
+        try {
+            updateTask(task, newTask);
+        } catch (DuplicateTaskException dte) {
+            throw new AssertionError("Modifying a task's tags only should not result in a duplicate. "
+                    + "See Task#equals(Object).");
+        }
+    }
+
+    /**
+     * Removes {@code tag} from all tasks in this {@code Organizer}.
+     */
+    public void removeTag(Tag tag) {
```

Comment:
Can we improve this code piece by adding assert(s) that we discussed in the tutorial?